src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue

```diff
@@ -64,59 +63,83 @@ export default defineComponent({
 
   setup(props) {
     const optionsListOrder = ref([])
+    const isLoading = ref(false)
     const saleOrderField = ref('')
+    const timeOut = ref(null)
 
     const isSalesTransaction = computed(() => {
-      const stringValue = store.getters.getInvoiceSearchQueryFilterByAttribute({
+      const stringValue = store.getters.getInvoicesQueryFilterByAttribute({
         containerUuid: props.uuidForm,
         attributeKey: 'is_sales_transaction'
       })
       return convertStringToBoolean(stringValue)
     })
 
+    const currentSaleOrder = computed({
+      set(newValue) {
+        console.log({ newValue })
+        store.commit('setInvoiceFieldQueryFilterByAttribute', {
+          containerUuid: props.uuidForm,
+          attributeKey: 'orderId',
+          value: newValue
+        })
+      },
+      get() {
+        return store.getters.getInvoicesQueryFilterByAttribute({
+          containerUuid: props.uuidForm,
+          attributeKey: 'orderId'
+        })
+      }
+    })
+
     const orderTitle = computed(() => {
       if (isSalesTransaction.value) {
         return lang.t('field.invoice.salesOrder')
       }
       return lang.t('field.invoice.purchaseOrder')
     })
 
-    function showList(isShow) {
-      if (isShow && isEmptyValue(optionsListOrder.value)) { filterSearchOrder({}) }
+    function showOrder(show) {
+      console.log(store.getters.getInvoceQueryFilters)
+      if (show && isEmptyValue(optionsListOrder.value)) remoteMethod()
     }
 
-    function filterSearchOrder(
-      searchQuery
-    ) {
-      requestListOrders({
-        search_value: searchQuery
-      })
-        .then(response => {
-          const { records } = response
-          optionsListOrder.value = records.map((list) => {
-            return {
-              ...list,
-              displayColumn: list.values.DisplayColumn
-            }
-          })
+    function remoteMethod(searchValue) {
+      clearTimeout(timeOut.value)
+      timeOut.value = setTimeout(() => {
+        isLoading.value = true
+        requestListOrders({
+          search_value: searchValue
         })
-    }
-
-    const currentValue = () => {
-      store.dispatch('searchInvociesInfos', {
-        order_id: saleOrderField.value
-      })
+          .then(response => {
+            const { records } = response
+            optionsListOrder.value = records.map((list) => {
```

Comment:
Remove loop to improve peformance https://github.com/solop-develop/frontend-core/blob/feature/legacy-support/src/components/ADempiere/FieldDefinition/FieldSearch/ProductInfo/PanelForm/QueryCriteria/productCategoryField.vue#L119

Reply:
https://github.com/solop-develop/frontend-core/pull/2220/commits/64508a3abac781fa837483fd62e7f96c800c9b13